### ISSUE TITLE
Deadlock between ServletContainerInitializer and Future.get from application code

### ISSUE THREAD

**Issue by njr-11** [OpenLiberty/open-liberty#11627]
There is a deadlock in webcontainer startup where a ServletContainerInitializer or ServletContextListener schedules a task using a managed executor and awaits completion of the task from the same method (onStartup/contextInitialized). After reproducing it in a test case (see commented TODO lines indicating disabled test code in https://github.com/OpenLiberty/open-liberty/pull/11608 ), I’ve narrowed it down to a deadlock between a webcontainer object, com/ibm/ws/webcontainer/webapp/WebApp$1 and the propagation of webcontainer metadata context to the thread where the task is attempting to start.

Here is that stack of the initialize thread, which obtains the lock, schedules a task and awaits the task result with Future.get(),

```
3XMTHREADINFO      "Default Executor-thread-14" J9VMThread:0x0000000007F15700, omrthread_t:0x00007FA26DAD15D8, java/lang/Thread:0x0000000701915498, state:P, prio=5
3XMJAVALTHREAD            (java/lang/Thread getId:0x48, isDaemon:true)
3XMTHREADINFO1            (native thread ID:0x4C153B, native priority:0x5, native policy:UNKNOWN, vmstate:P, vm thread flags:0x00020001)
3XMTHREADINFO2            (native stack address range from:0x00007000107DE000, to:0x000070001081E000, size:0x40000)
3XMCPUTIME               CPU usage total: 0.167381000 secs, current category="Application"
3XMTHREADBLOCK     Parked on: com/ibm/ws/threading/internal/PolicyTaskFutureImpl$State@0x000000070165F4A8 Owned by: <unknown>
3XMHEAPALLOC             Heap bytes allocated since last GC cycle=0 (0x0)
3XMTHREADINFO3           Java callstack:
4XESTACKTRACE                at jdk/internal/misc/Unsafe.park(Native Method)
4XESTACKTRACE                at java/util/concurrent/locks/LockSupport.park(LockSupport.java:194)
4XESTACKTRACE                at java/util/concurrent/locks/AbstractQueuedSynchronizer.parkAndCheckInterrupt(AbstractQueuedSynchronizer.java:885)
4XESTACKTRACE                at java/util/concurrent/locks/AbstractQueuedSynchronizer.doAcquireSharedInterruptibly(AbstractQueuedSynchronizer.java:1039)
4XESTACKTRACE                at java/util/concurrent/locks/AbstractQueuedSynchronizer.acquireSharedInterruptibly(AbstractQueuedSynchronizer.java:1345(Compiled Code))
4XESTACKTRACE                at com/ibm/ws/threading/internal/PolicyTaskFutureImpl.await(PolicyTaskFutureImpl.java:448)
4XESTACKTRACE                at com/ibm/ws/threading/internal/PolicyTaskFutureImpl.get(PolicyTaskFutureImpl.java:580)
4XESTACKTRACE                at concurrent/cdi2/web/ConcurrentCDI2ServletContainerInit.onStartup(ConcurrentCDI2ServletContainerInit.java:36)
4XESTACKTRACE                at com/ibm/ws/webcontainer/webapp/WebApp.initializeServletContainerInitializers(WebApp.java:2530)
4XESTACKTRACE                at com/ibm/ws/webcontainer/webapp/WebApp.initialize(WebApp.java:1027)
4XESTACKTRACE                at com/ibm/ws/webcontainer/webapp/WebApp.initialize(WebApp.java:6647)
5XESTACKTRACE                   (entered lock: com/ibm/ws/webcontainer/webapp/WebApp$1@0x0000000701BDC0C8, entry count: 1)
4XESTACKTRACE                at com/ibm/ws/webcontainer/osgi/DynamicVirtualHost.startWebApp(DynamicVirtualHost.java:467)
4XESTACKTRACE                at com/ibm/ws/webcontainer/osgi/DynamicVirtualHost.startWebApplication(DynamicVirtualHost.java:462)
4XESTACKTRACE                at com/ibm/ws/webcontainer/osgi/WebContainer.startWebApplication(WebContainer.java:1146)
4XESTACKTRACE                at com/ibm/ws/webcontainer/osgi/WebContainer.access$000(WebContainer.java:110)
4XESTACKTRACE                at com/ibm/ws/webcontainer/osgi/WebContainer$2.run(WebContainer.java:951)
4XESTACKTRACE                at com/ibm/ws/threading/internal/ExecutorServiceImpl$RunnableWrapper.run(ExecutorServiceImpl.java:239)
4XESTACKTRACE                at java/util/concurrent/Executors$RunnableAdapter.call(Executors.java:515(Compiled Code))
4XESTACKTRACE                at java/util/concurrent/FutureTask.run(FutureTask.java:264(Compiled Code))
4XESTACKTRACE                at java/util/concurrent/ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128(Compiled Code))
4XESTACKTRACE                at java/util/concurrent/ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
4XESTACKTRACE                at java/lang/Thread.run(Thread.java:825)
```

And here is the stack of the task thread,

```
3XMTHREADINFO      "Default Executor-thread-21" J9VMThread:0x0000000007F4BC00, omrthread_t:0x00007FA26F983450, java/lang/Thread:0x00000007018FF0F8, state:B, prio=5
3XMJAVALTHREAD            (java/lang/Thread getId:0x4F, isDaemon:true)
3XMTHREADINFO1            (native thread ID:0x4C1567, native priority:0x5, native policy:UNKNOWN, vmstate:B, vm thread flags:0x00000201)
3XMTHREADINFO2            (native stack address range from:0x00007000109B3000, to:0x00007000109F3000, size:0x40000)
3XMCPUTIME               CPU usage total: 0.433488000 secs, current category="Application"
3XMTHREADBLOCK     Blocked on: com/ibm/ws/webcontainer/webapp/WebApp$1@0x0000000701BDC0C8 Owned by: "Default Executor-thread-14" (J9VMThread:0x0000000007F15700, java/lang/Thread:0x0000000701915498)
3XMHEAPALLOC             Heap bytes allocated since last GC cycle=0 (0x0)
3XMTHREADINFO3           Java callstack:
4XESTACKTRACE                at com/ibm/ws/webcontainer/webapp/WebApp.initialize(WebApp.java:6645)
4XESTACKTRACE                at com/ibm/ws/webcontainer/osgi/WebContainerListener.initialize(WebContainerListener.java:211)
4XESTACKTRACE                at com/ibm/ws/container/service/metadata/internal/MetaDataServiceImpl.getMetaData(MetaDataServiceImpl.java:272)
4XESTACKTRACE                at com/ibm/ws/context/service/serializable/ThreadContextDescriptorImpl.taskStarting(ThreadContextDescriptorImpl.java:348(Compiled Code))
4XESTACKTRACE                at com/ibm/ws/concurrent/internal/TaskLifeCycleCallback.onStart(TaskLifeCycleCallback.java:223)
4XESTACKTRACE                at com/ibm/ws/threading/internal/PolicyTaskFutureImpl.run(PolicyTaskFutureImpl.java:706)
4XESTACKTRACE                at com/ibm/ws/threading/internal/PolicyExecutorImpl.runTask(PolicyExecutorImpl.java:1135)
4XESTACKTRACE                at com/ibm/ws/threading/internal/PolicyExecutorImpl$GlobalPoolTask.run(PolicyExecutorImpl.java:189)
4XESTACKTRACE                at java/util/concurrent/ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128(Compiled Code))
4XESTACKTRACE                at java/util/concurrent/ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
4XESTACKTRACE                at java/lang/Thread.run(Thread.java:825)
```
